KomaMRIFiles/src/Sequence/WritePulseq.jl

```diff
@@ -0,0 +1,407 @@
+"""
+    typeon_obj = get_typeon_obj(seq::Sequence, type::String)
+
+Returns the vector `typeon_obj` of objects (RF, Grad or ADC) where the object is on in a
+given sequence `seq` according to the `type` ∈ ["rf", "gr", "adc"] argument.
+"""
+function get_typeon_obj(seq::Sequence, type::String)
+    type == "rf"  && return [s.RF[1] for s ∈ seq if is_RF_on(s)]
+    type == "gr"  && return [g for s ∈ seq for g ∈ s.GR[:,1] if sum(abs.(g.A)) > 0]
+    type == "adc" && return [s.ADC[1] for s ∈ seq if is_ADC_on(s)]
+    return []
```

Comment:
Why are these functions not specialized depending on the type and use if statements?

Reply:
Function removed and replaced by `get_events_on` which is independet of the type. Addressed in 0b668ec